blazar/plugins/networks/network_plugin.py

```diff
@@ -789,3 +793,117 @@ def _filter_networks_by_properties(self, network_properties,
         else:
             return db_api.network_list()
+
+
+class NetworkMonitorPlugin(monitor.GeneralMonitorPlugin, neutron.NeutronClientWrapper):
+    """
+    Monitor plugin for network resources
+    """
+
+    def __new__(cls, *args, **kwargs):
+        return super(NetworkMonitorPlugin, cls).__new__(cls, *args, **kwargs)
+
+    def filter_allocations(self, reservation, resource_ids):
+        return [alloc for alloc
+                in reservation["network_allocations"]
+                if alloc["network_id"] in resource_ids]
+
+    def get_reservations_by_resource_ids(self, resource_ids,
+                                         interval_begin, interval_end):
+        return db_utils.get_reservations_by_floatingip_ids(resource_ids,
+                                                          interval_begin,
+                                                          interval_end)
+
+    def get_notification_event_types(self):
+        """Get event types of notification messages to handle."""
+        return ['service.update']
+
+    def notification_callback(self, event_type, payload):
+        return {}
+
+    def set_reservable(self, resource, is_reservable):
+        # network segments does not have reservable flag
+        LOG.debug("set_reservable() for network monitor is not implemented")
+        pass
+
+    def get_unreservable_resourses(self):
+        # because network segments does not have reservable flag
+        # there are no unreservable resources
+        return []
+
+    def poll_resource_failures(self):
+        failed = []
+        recovered = []
+        dry_run = CONF[plugin.RESOURCE_TYPE].enable_polling_monitor_dry_run
+        network_segments = db_api.network_list()
+
+        def process_network(network):
+            """
+            For each blazar network, check if the network is not in use by a reservation
+            and get the neutron network based on segment_id
+            Tear down the network, if neutron network is using a vlan when there is no active
+            reservation is blazar
+
+            1. Detach the network's subnets from the router.
+            2. Delete the subnet(s)
+            3. Delete the network
+
+            Args:
+                network (dict): blazar's network_segment row
+            """
+            network_id_from_blazar = network["id"]
+            segment_id = network["segment_id"]
+            # gets the most recent reservation for the blazar network
+            network_curr_reservation = db_utils.get_most_recent_reservation_info_by_network_id(network_id_from_blazar)
+            if network_curr_reservation and network_curr_reservation['status'] == status.reservation.ACTIVE:
+                # This means that network works fine as the reservation started fine
+                LOG.debug(f"Network {network_id_from_blazar} - VLAN {segment_id} is in active reservation {network_curr_reservation['id']} - skipping")
+                return
+            LOG.info(
```

Comment:
Is this log statement accurate? This code seems to be no reservation because of the return above. I do not think we want a log statement here, as this would print something every time the monitor runs for every network. Additionally, I think this may raise an error if `network_curr_reservation` is `None`.

Reply:
This is supposed to be a debug statement and yes this line breaks when it is None